src/core/main.c

```diff
@@ -73,6 +72,8 @@ static struct PyModuleDef core_module_def = {
   .m_size = -1,
 };
 
+int py_exec = Py_file_input;
+
```

Comment:
Is this still needed?

Reply:
Yes it is needed, I'll rearrange the code to make it more understandable.

Reply:
Okay I rearranged things a bit. Let me know if it looks better now.